### ISSUE TITLE
Allow filtering of server roles based on environment

### ISSUE THREAD

**Issue by dtjensen** [OctopusDeploy/Issues#531]
In my environment, I have some servers that host both dev and QA instances, but not for the same software components.  For example, the dev instance of service A runs on server C, and QA instance of service A runs on server D.  The dev instance of service B runs on server D and the QA instance of service B runs on server C.

In the dashboard, both servers C and D will show the service "A" and "B" roles in both the "dev" and "QA" environments, which is not accurate since the QA instance of service A is not on server C.  

My desired outcome would be that in the dev environment, only service A would show on server C and only service B would show on server D.  And in the QA environment, only service A would show on server D and only service B would show on server C.

Basically, we need a way to logically group the servers into environments that lays on top of the physical groupings.
